src/charm.py

```diff
@@ -193,6 +198,15 @@ def _on_config_changed(self, _: HookEvent) -> None:
         """
         self._configure_pod()
 
+    def _on_saml_data_available(self, event: SamlDataAvailableEvent) -> None:
+        """Handle SAML data available."""
+        fingerprint = hashlib.sha1(base64.b64decode(event.certificates[0])).hexdigest()  # nosec
+        relation = self.model.get_relation("saml")
+        # Will ignore union-attr since asserting the relation type will make bandit complain.
+        relation.data[self.app].update({"fingerprint": fingerprint})  # type: ignore[union-attr]
+        relation.data[self.app].update({"entity_id": event.entity_id})  # type: ignore[union-attr]
```

Comment:
Is that `entity_id` used?

Reply:
not anymore, thank you for spotting that!